### PR TITLE
[contrib:sidekiq] handle wrapped ActiveJob resource

### PR DESCRIPTION
**Pull request by ufoot**
This PR should fix https://github.com/DataDog/dd-trace-rb/issues/90 the fix is quite straightforward so it's mostly about testing it.

### CODE REVIEW

**Comment by ufoot** [comment]
Looks like there are conflicts with Ruby 1.9 branch. I probably need to rebase this one.

File: test/contrib/rails/rails_sidekiq_test.rb [hunk 1]
```diff
@@ -83,25 +96,58 @@ def perform(); end
...
     end
 
     assert_equal(1, spans.length)
     span = spans[0]
     assert_equal('sidekiq.job', span.name)
     assert_equal('RailsSidekiqTest::EmptyWorker', span.resource)
-    assert_match(/([0-9][a-f]){24}/, span.get_tag('sidekiq.job.id'))
-    assert_equal('true', 'sidekiq.job.retry')
+    assert_nil(span.get_tag('sidekiq.job.wrapper'))
+    assert_match(/([0-9]|[a-f]){24}/, span.get_tag('sidekiq.job.id'))
```
Comment (galdor) [inline test/contrib/rails/rails_sidekiq_test.rb]: `([0-9]|[a-f])` can be replaced by `[0-9a-f]`.

File: test/contrib/rails/rails_active_job_test.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    end
+
+    assert_equal(1, spans.length)
+    span = spans[0]
+    assert_equal('sidekiq.job', span.name)
+    assert_equal('RailsActiveJobTest::EmptyWorker', span.resource,
+                 'resource should be class doing the job')
+    assert_nil(span.get_tag('sidekiq.job.wrapper'))
+    assert_match(/([0-9]|[a-f]){24}/, span.get_tag('sidekiq.job.id'),
+                 'Job ID should be a 96-bit integer')
```
Comment (galdor) [inline test/contrib/rails/rails_active_job_test.rb]: `[0-9a-f]`.

File: test/contrib/rails/rails_sidekiq_test.rb [hunk 3]
```diff
@@ -73,4 +73,35 @@ def perform(); end
...
+  test 'Sidekiq middleware sends spans with the correct metadata' do
+    # configure Rails
+    update_config(:sidekiq_service, 'rails-sidekiq')
+
+    # add Sidekiq middleware
+    Sidekiq::Testing.server_middleware do |chain|
+      chain.add(Datadog::Contrib::Sidekiq::Tracer, tracer: @tracer)
+    end
+
+    sleep(1)
```
Comment (galdor) [inline test/contrib/rails/rails_sidekiq_test.rb]: Is that really necessary ? Is there a synchronization issue, and could it be solved without a random delay ? This is how we get flaky tests.

File: test/contrib/rails/rails_sidekiq_test.rb [hunk 4]
```diff
@@ -73,4 +73,35 @@ def perform(); end
...
+    @tracer.writer.spans() # empty test queue
+
+    # do something to force middleware execution
+    EmptyWorker.perform_async()
+
+    spans = []
+    30.times do
+      spans = @tracer.writer.spans()
+      break if spans
+      sleep(0.1)
```
Comment (galdor) [inline test/contrib/rails/rails_sidekiq_test.rb]: Same thing here, especially in a loop.

File: test/contrib/rails/rails_sidekiq_test.rb [hunk 5]
```diff
@@ -83,25 +96,58 @@ def perform(); end
...
+      break unless spans.empty?
+      sleep(0.1)
+    end
+
+    assert_equal(1, spans.length)
+    span = spans[0]
+    assert_equal('sidekiq.job', span.name)
+    assert_equal('RailsSidekiqTest::EmptyJob', span.resource)
+    assert_equal('TODO', span.resource, span.get_tag('sidekiq.job.wrapper'))
+    assert_match(/([0-9]|[a-f]){24}/, span.get_tag('sidekiq.job.id'))
```
Comment (galdor) [inline test/contrib/rails/rails_sidekiq_test.rb]: `[0-9a-f]`

File: test/contrib/rails/rails_sidekiq_test.rb (hunk 3 above)
Comment (ufoot) [inline test/contrib/rails/rails_sidekiq_test.rb]: Doh.... that one is a test stuff left here for no valuable reason.

**Pull request merged**